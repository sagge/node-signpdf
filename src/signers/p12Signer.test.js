import SignPdfError from '../SignPdfError';
import P12Signer from './p12Signer';

describe('Test signing', () => {
    it('expects P12 certificate to be Buffer', () => {
        try {
            // eslint-disable-next-line no-new
            new P12Signer('non-buffer');
            expect('here').not.toBe('here');
        } catch (e) {
            expect(e instanceof SignPdfError).toBe(true);
            expect(e.type).toBe(SignPdfError.TYPE_INPUT);
            expect(e.message).toMatchSnapshot();
        }
    });
    it('expects pdf to be Buffer', async () => {
        try {
            const signer = new P12Signer(Buffer.from(''));
            await signer.sign('non-buffer');
            expect('here').not.toBe('here');
        } catch (e) {
            expect(e instanceof SignPdfError).toBe(true);
            expect(e.type).toBe(SignPdfError.TYPE_INPUT);
            expect(e.message).toMatchSnapshot();
        }
    });
});
