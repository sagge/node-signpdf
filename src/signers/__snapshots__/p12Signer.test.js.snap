// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test signing expects P12 certificate to be Buffer 1`] = `"p12 certificate expected as Buffer."`;

exports[`Test signing expects pdf to be Buffer 1`] = `"PDF expected as Buffer."`;
